### ISSUE TITLE
Tokenization differs for different intepreter instances

### ISSUE THREAD

**Issue by pglock** [huggingface/transformers#2027]
## 🐛 Bug

Tokenization of `" "` changes for each python interpreter instance.

## To Reproduce

```python
from transformers import BertTokenizer

tokenizer = BertTokenizer.from_pretrained('bert-base-uncased')
for i in range(5):
    print(tokenizer.encode(" "))
```

## Environment

* Python version: 3.7.2
* PyTorch version: 1.3.1
* PyTorch Transformers version (or branch): 2.2.0

**Comment by TheEdoardo93** [comment]
```
> from transformers import BertTokenizer
> tokenizer = BertTokenizer.from_pretrained('bert-base-uncased')
> for i in range(5):
    print(tokenizer.encode(" "))
>>> [101, 100, 102]
[101, 100, 102]
[101, 100, 102]
[101, 100, 102]
[101, 100, 102]
```

I don't understand your question very much. Are you saying that with Python v.X the output of the code you've posted is different from that with Python v.Y?

> ##  Bug
> Tokenization of `" "` changes for each python interpreter instance.
> 
> ## To Reproduce
> ```python
> from transformers import BertTokenizer
> 
> tokenizer = BertTokenizer.from_pretrained('bert-base-uncased')
> for i in range(5):
>     print(tokenizer.encode(" "))
> ```
> 
> ## Environment
> * Python version: 3.7.2
> * PyTorch version: 1.3.1
> * PyTorch Transformers version (or branch): 2.2.0

**Comment by pglock** [comment]
No, if I execute this code in a script, the output differs each time.
For example:
```
# first run
[101, 0, 102]
[101, 0, 102]
[101, 0, 102]
[101, 0, 102]
[101, 0, 102]

# second run
[101, 100, 102]
[101, 100, 102]
[101, 100, 102]
[101, 100, 102]
[101, 100, 102]
```

**Comment by TheEdoardo93** [comment]
I've noticed this bug right now.

**1st run:**
```
> from transformers import BertTokenizer
> tokenizer = BertTokenizer.from_pretrained('bert-base-uncased')
> for i in range(5):
...     print(tokenizer.encode(" "))
>>> [101, 100, 102]
[101, 100, 102]
[101, 100, 102]
[101, 100, 102]
[101, 100, 102]
```

**2nd run:**
```
> from transformers import BertTokenizer
> tokenizer = BertTokenizer.from_pretrained('bert-base-uncased')
> for i in range(5):
...     print(tokenizer.encode(" "))
>>> [101, 103, 102]
[101, 103, 102]
[101, 103, 102]
[101, 103, 102]
[101, 103, 102]
```

Sometimes, the whitespace character " " is encoded with token with ID=100, other times with token with ID= 103. Why this behavior?

> No, if I execute this code in a script, the output differs each time.
> For example:
> 
> ```
> # first run
> [101, 0, 102]
> [101, 0, 102]
> [101, 0, 102]
> [101, 0, 102]
> [101, 0, 102]
> 
> # second run
> [101, 100, 102]
> [101, 100, 102]
> [101, 100, 102]
> [101, 100, 102]
> [101, 100, 102]
> ```

**Comment by thomwolf** [comment]
Feel like this is a non-deterministic behavior arising from encoding an empty sentence.

Do you have a real-world use case for encoding empty sentence?

**Comment by pglock** [comment]
I guess we could catch this case before using the model, but a deterministic behaviour would still be neat. 

If `" "` is the same as `""`, the tokenizer should just return an empty list right?

**Comment by pglock** [comment]
fixed in #2081
